upgrade-2.x.sh

```diff
@@ -249,6 +250,176 @@ function device_type_match {
+            systemctl stop docker-host
+            log "Clean inactive partition"
+            rm -rf "${sysroot:?}/"*
+            systemctl start docker-host
+            sleep 10
+        fi
+    fi
+
+    log "Starting hostapp-update"
+    hostapp-update -i "${update_package}" || log ERROR "hostapp-update has failed..."-
+}
+
+function non_hostapp_to_hostapp_update {
+    local update_package=$1
+    local sysroot="/mnt/sysroot/inactive"
+    local temp_sysroot
+
+    find_partitions
+    stop_services
+
+    # Mount spare root partition
+    umount "${update_part}" || true
+    mkfs.ext4 -F -L "${update_label}" "${update_part}"
+    temp_sysroot=$(mktemp -d)
+    mount "${update_part}" "${temp_sysroot}" || log ERROR "Cannot mount inactive partition ${update_part} to ${temp_sysroot}..."
+
+    case "${SLUG}" in
+        raspberry*)
+            log "Running pre-update fixes for ${SLUG}"
+            pre_update_pi_bootfiles_removal
+            ;;
+        *)
+            log "No device-specific pre-update fix for ${SLUG}"
+    esac
+
+    docker pull "${update_package}" || log ERROR "Couldn't pull docker image..."
+    mkfifo /tmp/resinos-image.docker
+    docker save "${update_package}" > /tmp/resinos-image.docker &
+    mkdir -p /mnt/data/resinhup/tmp
+
+    log "Starting hostapp-update"
+    # Note that the following docker daemon is started with a different --bip and --fixed-cidr
+    # setting, otherwise it is clashing with the system docker on resinOS >=2.3.0 || <2.5.1
+    # and then docker pull would not succeed
+    # The requrired volumes are:
+    #  * /mnt/sysroot/inactive: the new rootfs
+    #  * /mnt/boot: the boot partition
+    #  * /tmp/resinos-image: the FIFO file that is used to transfer the resinOS image into the container
+    #  * /mnt/data/resinhup/tmp: a location to use as docker temporary directory within the container
+    # shellcheck disable=SC2016
+    docker run \
+      --rm \
+      --name resinhup \
+      --privileged \
+      -v "${temp_sysroot}":/mnt/sysroot/inactive \
+      -v /mnt/boot:/mnt/boot \
+      -v /tmp/resinos-image.docker:/resinos-image.docker \
+      -v /mnt/data/resinhup/tmp:/mnt/data/resinhup/tmp \
+      "${update_package}" \
+      /bin/bash -c 'storage_driver=$(cat /boot/storage-driver) ; DOCKER_TMPDIR=/mnt/data/resinhup/tmp/ dockerd --log-driver=journald -s=$storage_driver --data-root='"${sysroot}"'/docker -H unix:///var/run/docker-host.sock --pidfile=/var/run/docker-host.pid --exec-root=/var/run/docker-host --bip 10.114.201.1/24 --fixed-cidr=10.114.201.128/25 --iptables=false & sleep 10; echo "Starting hostapp-update"; hostapp-update -f /resinos-image.docker' \
+    || log ERROR "Update based on hostapp-update has failed..."
+}
+
+function find_partitions {
+    # Find which partition is / and which we should write the update to
+    root_part=$(findmnt -n --raw --evaluate --output=source /)
+    log "Found root at ${root_part}..."
+    case $root_part in
+        *2)
```

Comment:
This index based approach will not work on boards like the edison or the jetson tx2 that do not have their root partition on offsets 2 or 3

Reply:
This function is only used for pre-hostapp-update devices, so for the Jetson TX2 it should not cause issue. For the Edison this will be an issue, when adding support for HUP in the near future. In that case will add Edison-specific stuff in this logic.